pkg/apis/apps/validation/validation.go

```diff
@@ -625,9 +633,12 @@ func ValidateDeploymentStatus(status *apps.DeploymentStatus, fldPath *field.Path
 
 // ValidateDeploymentUpdate tests if an update to a Deployment is valid.
 func ValidateDeploymentUpdate(update, old *apps.Deployment, opts apivalidation.PodValidationOptions) field.ErrorList {
-	allErrs := apivalidation.ValidateObjectMetaUpdate(&update.ObjectMeta, &old.ObjectMeta, field.NewPath("metadata"))
-	allErrs = append(allErrs, ValidateDeploymentSpec(&update.Spec, field.NewPath("spec"), opts)...)
-	return allErrs
+	updateClone := update.DeepCopy()
```

Comment:
Only the update Spec needs to be deep copied

Reply:
done